source/common/filter/tcp_proxy.h

```diff
@@ -30,7 +30,11 @@ namespace Filter {
   COUNTER(downstream_cx_tx_bytes_total)                                                            \
   GAUGE  (downstream_cx_tx_bytes_buffered)                                                         \
   COUNTER(downstream_cx_total)                                                                     \
-  COUNTER(downstream_cx_no_route)
+  COUNTER(downstream_cx_no_route)                                                                  \
+  COUNTER(upstream_pause_reading)                                                                  \
```

Comment:
It would be better IMO if these were defined as upstream cluster stats (see the link I originally referenced). They can then also be used in the HTTP flow and for other proxy scenarios.

P.S. please also document the new stats in the relevant parts of the docs.